.github/workflows/dev-docs.yml

```diff
@@ -0,0 +1,55 @@
+name: dev-docs 
+on:
+  push:
+    branches-ignore:
+      - master
+jobs: 
+  build: 
+    name: Build and Deploy Documentation 
+    runs-on: ubuntu-latest 
+    steps: 
+      - name: Checkout Master 
+        uses: actions/checkout@v2 
+      - name: Set up Python 3.7 
+        uses: actions/setup-python@v2 
+        with: 
+          python-version: '3.7.5' 
+      - name: Install dependencies 
+        run: | 
+          python -m pip install --upgrade pip 
+          pip install .
+          pip install mkdocs mkdocs-material mkdocstrings mkdocs-versioning
+          pip install typer-cli
+      - name: Set up mkdocs config
+        run: | 
+          export VERSION=dev
+          echo "Making docs for version $VERSION"
+          (envsubst '${VERSION}' < mkdocs-template.yml) > mkdocs.yml
+      - name: Generate CLI docs and copy CHANGELOG   
+        run: |
+          typer xnat_tools/xnat2bids.py utils docs --name xnat2bids --output docs/xnat2bids.md
+          typer xnat_tools/dicom_export.py utils docs --name xnat-dicom-export --output docs/dicom_export.md
+          typer xnat_tools/run_heudiconv.py utils docs --name xnat-heudiconv --output docs/run_heudiconv.md
+          typer xnat_tools/bids_postprocess.py utils docs --name bids-postprocess --output docs/bids_postprocess.md
+          cp CHANGELOG.md docs/changelog.md
+      - name: Push to remote
+        run: |
+          echo "Test push commit to the remote in the workflow" >> test.txt
+
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+          git add -A
+          git commit -m "docs: Update CLI and CHANGELOG.md"
```

Comment:
Fun fact, if you have two `-m` flags it'll use the first as the title and the second as the body of your commit message!